src/lib/Server.js

```diff
@@ -335,22 +335,17 @@ module.exports = class Server {
       });
     };
 
-    // Prometheus Metrics API
-    const routerPrometheusMetrics = createRouter();
-    app.use(routerPrometheusMetrics);
-
     // Check Prometheus credentials
     app.use(
       fromNodeMiddleware((req, res, next) => {
         if (!requiresPrometheusPassword || !req.url.startsWith('/metrics')) {
           return next();
         }
         const user = basicAuth(req);
-        if (requiresPrometheusPassword && !user) {
```

Comment:
Why did you remove this?

Reply:
This is unnecessary as there is already a check above

if (!requiresPrometheusPassword || !req.url.startsWith(‘/metrics’))

Reply:
OUCH. Although no!!!

Reply:
Your right